src/routes/v2/admin.rs

```diff
@@ -16,10 +19,43 @@ pub fn config(cfg: &mut web::ServiceConfig) {
     cfg.service(
         web::scope("admin")
             .service(count_download)
+            .service(add_minos_user)
+            .service(get_legacy_account)
             .service(process_payout),
     );
 }
 
+// Adds a Minos user to the database
+// This is an internal endpoint, and should not be used by applications, only by the Minos backend
+#[post("_minos-user-callback", guard = "admin_key_guard")]
+pub async fn add_minos_user(
+    minos_user: web::Json<MinosNewUser>, // getting directly from Kratos rather than Minos, so unparse
+    client: web::Data<PgPool>,
+) -> Result<HttpResponse, ApiError> {
+    let minos_new_user = minos_user.into_inner();
+    let mut transaction = client.begin().await?;
+    link_or_insert_new_user(&mut transaction, minos_new_user).await?;
+    transaction.commit().await?;
+    Ok(HttpResponse::Ok().finish())
+}
+
+#[get("_legacy_account/{github_id}", guard = "admin_key_guard")]
+
+pub async fn get_legacy_account(
+    client: web::Data<PgPool>,
+    github_id: web::Path<i32>,
+) -> Result<HttpResponse, ApiError> {
+    println!("Here :)");
+    let github_id = github_id.into_inner();
+    println!("Github id: {}", github_id);
+    let mut transaction = client.begin().await?;
+    let user = user_item::User::get_from_github_id(github_id as u64, &mut *transaction).await?;
+    let user: Option<User> = user.map(|u| u.into());
+    println!("User found");
+    transaction.commit().await?;
+    Ok(HttpResponse::Ok().json(user))
+}
```

Comment:
print statements

Reply:
also why is there a transaction for get legacy account? can't it just use the pool directly